packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -1618,6 +1618,220 @@ export default{
+        enabled: false,
+      });
+      mockPublishRoutesRequest({
+        routes: ["http://example.com/*"],
+      });
+      await runWrangler("publish index.js");
+
+      expect(std.out).toMatchInlineSnapshot(`
+        "Uploaded test-name (TIMINGS)
+        Published test-name (TIMINGS)
+          test-name.test-sub-domain.workers.dev
+          http://example.com/*"
+      `);
+      expect(std.err).toMatchInlineSnapshot(`""`);
+      expect(std.warn).toMatchInlineSnapshot(`""`);
+    });
+
+    it("can deploy to both workers.dev and routes if both defined (environments: 1)", async () => {
+      writeWranglerToml({
+        workers_dev: true,
+        env: {
+          production: {
+            routes: ["http://production.example.com/*"],
+          },
+        },
+      });
+      writeWorkerSource();
+      mockSubDomainRequest();
+      mockUploadWorkerRequest({ env: "production", legacyEnv: true });
+      mockUpdateWorkerRequest({
+        enabled: false,
+        env: "production",
+        legacyEnv: true,
+      });
+      mockPublishRoutesRequest({
+        routes: ["http://production.example.com/*"],
+        env: "production",
+        legacyEnv: true,
+      });
+      await runWrangler("publish index.js --env production");
+
+      expect(std.out).toMatchInlineSnapshot(`
+        "Uploaded test-name-production (TIMINGS)
+        Published test-name-production (TIMINGS)
+          test-name-production.test-sub-domain.workers.dev
+          http://production.example.com/*"
+      `);
+      expect(std.err).toMatchInlineSnapshot(`""`);
+      expect(std.warn).toMatchInlineSnapshot(`""`);
+    });
+
+    it("can deploy to both workers.dev and routes if both defined (environments: 1)", async () => {
+      writeWranglerToml({
+        env: {
+          production: {
+            workers_dev: true,
+            routes: ["http://production.example.com/*"],
+          },
+        },
+      });
+      writeWorkerSource();
+      mockSubDomainRequest();
+      mockUploadWorkerRequest({ env: "production", legacyEnv: true });
+      mockUpdateWorkerRequest({
+        enabled: false,
+        env: "production",
+        legacyEnv: true,
+      });
+      mockPublishRoutesRequest({
+        routes: ["http://production.example.com/*"],
+        env: "production",
+        legacyEnv: true,
+      });
+      await runWrangler("publish index.js --env production");
+
+      expect(std.out).toMatchInlineSnapshot(`
+        "Uploaded test-name-production (TIMINGS)
+        Published test-name-production (TIMINGS)
+          test-name-production.test-sub-domain.workers.dev
+          http://production.example.com/*"
+      `);
+      expect(std.err).toMatchInlineSnapshot(`""`);
+      expect(std.warn).toMatchInlineSnapshot(`""`);
+    });
+
+    it("will deploy only to routes when workers_dev is false (environments) ", async () => {
+      writeWranglerToml({
+        env: {
+          production: {
+            workers_dev: false,
```

Comment:
Can we have one more test where `workers_dev` is false at the top level and then there is a route in an environment. In that case it should inherit the `false` from the top level and only deploy to the route.